tests/native/scaled.cpp

```diff
@@ -1,14 +1,37 @@
+#include "gtest/gtest.h"
```

Comment:
Also there seems to be a problem when the element value and scaling factor are both of `int8_t`.  The element type of the scaled mdspan seems to be `int` not `int8_t`.
```
  TEST(scaled, mdspan_int8)
  {
    using vector_element_type = std::int8_t;
    using scaling_factor_type = std::int8_t;

    using vector_t =
    mdspan<vector_element_type, extents<dynamic_extent>>;

    constexpr std::size_t vectorSize (5);
    constexpr std::size_t storageSize = vectorSize;
    std::vector<vector_element_type> storage (storageSize);

    vector_t x (storage.data (), vectorSize);

    for (std::size_t k = 0; k < vectorSize; ++k) {
      const vector_element_type x_k = vector_element_type(k) + 1.0;
      x(k) = x_k;
    }

    const scaling_factor_type scalingFactor (-3.0);
    auto x_scaled = scaled (scalingFactor, x);

    using val_t = typename decltype(x_scaled)::value_type;
    //static_assert( std::is_same_v< std::remove_cv_t<val_t>, int > );
    static_assert( std::is_same_v< std::remove_cv_t<val_t>, vector_element_type > );
  }
```

Reply:
@youyu3 The implementation is following the usual integer promotion rules.

https://godbolt.org/z/Ec3zeGqP9
```c++
#include <type_traits>
#include <cstdint>

int main()
{
    int8_t scalingFactor = 3.0; // narrowed
    int8_t x = 4;
    auto d_times_i = scalingFactor * x; // what scaled_scalar does
    static_assert(std::is_same_v<decltype(d_times_i), int>);  
    return 0;
}
```

Reply:
Ah, good to know that integer promotion rules are designed in that way.  This makes sense, but I'll need to fix my code.

Reply:
Yup, it's a little weird that the product of two `int8_t` is `int`, but such is the weird language we find ourselves using ; - )